ee/tools/mdm/README.md

```diff
@@ -0,0 +1,28 @@
+## MDM Push CSR generation tool
+
+### Build
+
+Build like any other go program:
+
+``` sh
+go build -o mdm-gen-cert .
+```
+
+### Usage
+
+The following environment variables must be configured:
+
+`VENDOR_CERT_PEM` - Fleet's MDM Vendor certificate in PEM format.
+`VENDOR_KEY_PEM` - Fleet's MDM Vendor private key in PEM format.
+`VENDOR_KEY_PASSPHRASE` - Passphrase for the MDM Vendor private key.
+`CSR_BASE64` - Base64 encoded CSR submitted from the Fleet server or `fleetctl` on behalf of the user. (Note: this is
+accepted as an environment variable to mitigate against command injection attacks from untrusted user input.)
+
+The program outputs the email from the signing request, and the signed request as JSON. For example:
+
+```json
+{"email":"fleetuser@example.com","request":"PD94bWw..."}
+```
+
+The email should be validated against the email denylist, and then the request contents should be
+sent to that email address as an attachment (eg. `apple-apns-request.txt`).
```

Comment:
Out of curiosity: Does Apple accept the `.txt` extension?

MicroMDM uses the `.plist` extension (https://github.com/micromdm/micromdm/blob/main/docs/user-guide/quickstart.md).

Reply:
this is a good point, from the documentation of `*PushCertificateRequest.Encode()`:

> Encode marshals a PushCertificateRequest to an XML Plist file and returns a base64 encoded byte representation of the request.

leaving a suggestion in case it's easier to make the change:

```suggestion
sent to that email address as an attachment (eg. `apple-apns-request.plist`).
```

Reply:
Apple seems to accept any extension (I've tested `.txt` and `.foobar`). The reasons why I went with `.txt` are

1) It's not actually a plist, it's a base64 encoded plist.
2) I wonder if the `.txt` extension is less likely to be caught up in email spam filters? The mdmcert.download emails were always being automatically quarantined by gmail.

Do y'all still think it should be `.plist`? I'm definitely open to it.

Reply:
given that Apple accepts any extension, it makes sense to me to optimize for email delivery, we can adjust later if need to be, +1 from me

Reply:
+1!